core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -454,14 +454,20 @@ private[spark] class BlockManager(
       case Some(info) =>
         val level = info.level
         logDebug(s"Level for block $blockId is $level")
+        val taskAttemptId = Option(TaskContext.get()).map(_.taskAttemptId())
```

Comment:
This is getting the `taskAttemptId` from the main thread, in case a TaskContext is not propagated properly to all child threads for the task, we would fail in getting the `taskAttemptId` in `BlockInfoManager`, see the test case added in this PR.